edx_management_commands/management_commands/management/commands/manage_user.py

```diff
@@ -0,0 +1,131 @@
+                    'addresses do not match.'
+                ).format(user.username)
+            )
+
+    def _handle_remove(self, username, email):
+
+        users = get_user_model().objects.filter(username=username)
+        if not users:
+            self.stderr.write(_('Did not find a user with username "{}" - skipping.').format(username))
+            return
+
+        user = users[0]
+        self._check_email_match(user, email)
+        self.stderr.write(_('Removing user: "{}"').format(user))
+        user.delete()
+
+    def handle(self, username, email, is_remove, is_staff, is_superuser, groups, *args, **options):
+
+        if is_remove:
+            return self._handle_remove(username, email)
+
+        old_groups, new_groups = [], []
+        user, created = get_user_model().objects.get_or_create(username=username)
+
+        if created:
+            user.set_unusable_password()
+            self.stderr.write(_('Created new user: "{}"').format(user))
+            self._maybe_update(user, 'email', email)
+        else:
+            # NOTE, we will not update the email address of an existing user.
+            self.stderr.write(_('Found existing user: "{}"').format(user))
+            self._check_email_match(user, email)
+            old_groups = list(user.groups.all())
+
+        self._maybe_update(user, 'is_staff', is_staff)
+        self._maybe_update(user, 'is_superuser', is_superuser)
+
+        # resolve the specified groups
+        for group_name in groups or []:
+
+            try:
+                group = Group.objects.get(name=group_name)  # pylint: disable=no-member
+                new_groups.append(group)
+            except Group.DoesNotExist:
+                # warn, but move on.
+                self.stderr.write(_('Could not find a group named "{}" - skipping.').format(group_name))
+
+        add_groups = [p for p in new_groups if p not in old_groups]
+        remove_groups = [p for p in old_groups if p not in new_groups]
+
+        if add_groups:
+            for ag in add_groups:
+                self.stderr.write(
+                    _(
+                        'Adding user "{username}" to group "{group_name}"'
+                    ).format(
+                        username=username, group_name=ag.name
+                    )
+                )
+                user.groups.add(ag)
+        else:
+            self.stderr.write(_('Not adding user "{}" to any groups').format(username))
+
+        if remove_groups:
+            for rg in remove_groups:
+                self.stderr.write(
+                    _(
+                        'Removing user "{username}" from group "{group_name}"'
+                    ).format(
+                        username=username, group_name=rg.name
+                    )
+                )
+                user.groups.remove(rg)
+        else:
+            self.stderr.write(_('Not removing user "{}" from any groups').format(username))
```

Comment:
For bonus Python points you could use `for`/`else`: https://docs.python.org/2/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops